### DISCUSSION TITLE
Swiper Element - Breakpoints

### DISCUSSION THREAD

**Discussion by okzea** [nolimits4web/swiper#6449] [Q&A] [answered]
Hey there, I'm trying to setup breakpoints in web component attributes but I'm having this error:
![image](https://user-images.githubusercontent.com/10658943/222433007-880f535e-d250-44ed-80b9-bf1da24567ee.png)

Here's the html snippet:
`<swiper-container slides-per-view="1" breakpoints="{768: {slidesPerView: 3}}">`

Does anyone have an idea of what I'm doing wrong or if that's just a bug?

Thanks 🙏

**Comment by Koreanderson** [comment]
I'm also curious to see if this is possible with only web component attributes.

All Swiper examples using using breakpoints on the web component are re-initializing it by passing in a breakpoints prop.

It'd be really helpful if the web component could parse the breakpoint attribute properly as json.

**Comment by justrustyspoons** [comment]
In Vue I have this solution...

Assign the ref as a function https://vuejs.org/guide/essentials/template-refs.html#function-refs

```
<swiper-container
    init="false"
    :ref="(el) => setSwiperEl(el)"
>
```
then in the setup script after calling ```register()```
```
const swiperEl = ref(null);
const swiperParams = {
    slidesPerView: 1,
    spaceBetween: 20,
    breakpoints: {
        1024: {
            slidesPerView: 2,
            spaceBetween: 10,
        },
        1720: {
            slidesPerView: 3,
            spaceBetween: 20,
        },
    },
};
const setSwiperEl = (el) => {
    swiperEl.value = el;
    Object.assign(swiperEl.value, swiperParams);
    swiperEl.value.initialize();
};

**Comment by asif-jalil** [comment]
I am also having a problem on react. Has anyone found any solution?

**Comment by asif-jalil** [comment]
@okzea did you find any solution?

**Reply by okzea** [reply]
@asif-jalil unfortunately no, I went with inline script...

**Comment by constmoon** [comment]
I'm facing the same issue, is there any updates?

**Reply by asif-jalil** [reply]
@nolimits4web please see that.

**Comment by Akistapace** [comment]
I made it using react and setting that way, using JSON.stringify and it works well
 
 ```
<swiper-container slides-per-view="2"  breakpoints={JSON.stringify({ 768: { slidesPerView: 3 } })}> 
 ....
 </swiper-container>
```

**Reply by asif-jalil** [reply]
That's really nice

**Reply by kirgans** [reply]
don't need JSON.stringify here 

`const breakpoints = {
    '@1.33': {
        slidesPerView: 6.5
    },
    320: {
        slidesPerView: 4.5
    },
    1200: {
        slidesPerView: 6.5
    }
};

<Swiper
      breakpoints={breakpoints}
      ....
      >
      <SwiperSlide></SwiperSlide>
</Swiper>`

And follow Swiper and SwiperSlide components implementation here : https://swiperjs.com/blog/using-swiper-element-in-react

**Reply by Akistapace** [reply]
@spacedev75  It was necessary in my case as I was using the syntax <swiper-container> instead of <Swiper>

**Comment by okzea** [comment, marked as answer]
Need to use params as prop according to https://github.com/nolimits4web/swiper/issues/6488

**Reply by okzea** [reply]
It has been fixed in v10 https://swiperjs.com/blog/swiper-v10#json-in-attributes
